tools/data-handler/src/validate.ts

```diff
@@ -95,7 +99,120 @@ export class Validate {
 
   // Return full path and filename.
   private fullPath(file: Dirent): string {
-    return join(file.path, file.name);
+    return join(file.parentPath, file.name);
+  }
+
+  // Handles validating .schema files
+  private async validateSchemaFiles(files: Dirent[]) {
+    const schema = this.validator.schemas[Validate.dotSchemaSchemaId];
+
+    if (!schema) {
+      throw new Error(`.schema schema not found`);
+    }
+
+    const message: string[] = [];
+
+    for (const file of files) {
+      const fullPath = this.fullPath(file);
+
+      const result = this.validator.validate(
+        await readJsonFile(fullPath),
+        schema,
+      );
+      for (const error of result.errors) {
+        const msg = `Validation error from '${fullPath}': ${error.message}.`;
+        message.push(msg);
+      }
+    }
+    return message;
+  }
+
+  // Handles reading and validating 'contentSchema' in a directory.
+  private async readAndValidateContentFiles(
```

Comment:
This function could be before `validateSchemaFiles` so that `private` functions are in order